internal/controller/clusterdeployment_controller.go

```diff
@@ -312,8 +312,9 @@ func (r *ClusterDeploymentReconciler) updateCluster(ctx context.Context, mc *hmc
 		},
 		ChartRef: clusterTpl.Status.ChartRef,
 	}
-	if clusterTpl.Spec.Helm.ChartSpec != nil {
-		hrReconcileOpts.ReconcileInterval = &clusterTpl.Spec.Helm.ChartSpec.Interval.Duration
+	reconcileInterval := clusterTpl.Spec.Helm.ChartSpec.Interval.Duration
+	if reconcileInterval != 0 {
+		hrReconcileOpts.ReconcileInterval = &reconcileInterval
```

Comment:
Could you please remove this change? It seems to have been unintentionally introduced during the rebase process.